### PR TITLE
Check `FLAG_POPUP` to close an AcceptDialog when parent is focused

### LINKED ISSUE #79150: `AcceptDialog.close_on_escape` changes how mouse clicks are interpreted
### Godot version

v4.2.dev.custom_build [443820686]

### System information

Debian Linux X11 Xfce

### Issue description

`AcceptDialog.dialog_close_on_escape` has as description:
```
If [code]true[/code], the dialog will be hidden when the escape key ([constant KEY_ESCAPE]) is pressed.
```
However the parameter also affects, how mouse clicks outside of the popup are interpreted.

https://github.com/godotengine/godot/blob/443820686cad24bbdd7e7050fa3d3f33f43a7bb5/scene/gui/dialogs.cpp#L47-L51

I would expect, that the behavior and the documentation fits.

It is unclear, if the documentation is wrong or if this is a bug and `close_on_escape` should not have an impact on mouse clicks. This got introduced in #59764.

I am inclined to say, that the documentation is correct, because it fits better to the name of the parameter.

### Steps to reproduce

1. Load and start MRP
2. Click button to create Popup
3. Click outside of Popup (left popup closes, right popup doesn't close)

### Minimal reproduction project

[PopupEscapeClose.zip](https://github.com/godotengine/godot/files/11981263/PopupEscapeClose.zip)

### PR DESCRIPTION
**Pull request by ItsNL**
Fix #79150 

The `close_on_escape` property was used to be able to close an AcceptDialog by pressing the `Esc` key, but it was also checked to close when the parent was focused, meaning it had lost the focus. 

With this pull request instead of checking if `close_on_escape` is `true` to close automatically when the parent is focused (For example clicking outside of the dialog), it will check if `popup_window` is `true`.

Now `dialogue_close_on_escape`  only affects if pressing the `Esc` key will close the dialog.

Here is a video showing the change:

https://github.com/godotengine/godot/assets/109873547/afef6078-7c84-40d9-b76a-1e77f827b2e6

***Note:** When using `popup_window` you will need that `exclusive` is `false` so that it works. `exclusive` as `true` blocks all input that is outside the dialog.*
***Note 2:** When only using `dialog_close_on_escape` clicking outside will make the dialog's focus be lost and so using `Esc` won't work until the dialog has the focus again.*

### CODE REVIEW

**Comment by Sauermann** [comment]
I was rather thinking along the lines of
``` diff
diff --git a/scene/gui/dialogs.cpp b/scene/gui/dialogs.cpp
index 0860bf3968..5865e06ec9 100644
--- a/scene/gui/dialogs.cpp
+++ b/scene/gui/dialogs.cpp
@@ -45,7 +45,7 @@ void AcceptDialog::_input_from_window(const Ref<InputEvent> &p_event) {
 }
 
 void AcceptDialog::_parent_focused() {
-       if (close_on_escape && !is_exclusive()) {
+       if (!is_exclusive()) {
                _cancel_pressed();
        }
 }
```
I am not sure, if there is a use-case supporting `close_on_unfocused`.

**Comment by ItsNL** [comment]
Your option uses `exclusive` as your way to decide if the dialog will close or not when clicking outside. But, when enabling `exclusive` you can't interact with the rest of the interface.
So, a use-case of `close_on_unfocused` would be having it off, but being able to have the entire UI accessible without having to close the dialog first or closing it automatically. With the option of still being able to close it using Esc (if wanted).
In conclusion, with that option, a user has to decide whether they allow clicking outside of the dialog to close it or not allow interacting with any other UI at all.

**Review by bruvzg — COMMENTED** [review]
It won't work with native windows when switching focus to the other app, so the property name can be misleading. For native windows, the same behavior is handled by FLAG_POPUP, so it probably should check it instead of adding a new property.

**Comment by ItsNL** [comment]
I have the changes ready using FLAG_POPUP and it works fine. Should I create another Pull Request and close this one? Because now it has nothing to do with this Pull Request.

**Comment by KoBeWi** [comment]
You can change commit message and title. No need to make new PR.

**Comment by Sauermann** [comment]
I can confirm, that this solves the linked issue for me.

After a few tests, I believe, that the note can be removed:
```
[b]Note:[/b] This property only works with native windows.
```
The description fits with the behavior of non-native windows.

**Comment by ItsNL** [comment]
Ok. I've deleted the note now.

**Review by Sauermann — APPROVED** [review]
Have tested it successfully with my MRP in the linked issue.

**Comment by akien-mga** [comment]
Thanks!
